Core/include/Acts/EventData/VectorTrackContainer.hpp

```diff
@@ -88,6 +80,23 @@ class VectorTrackContainerBase {
     }
   }
 
+  void checkConsistency() const {
+    size_t size = m_tipIndex.size();
+    (void)size;
+
+    assert(m_tipIndex.size() == size);
+    assert(m_params.size() == size);
+    assert(m_cov.size() == size);
+    assert(m_referenceSurfaces.size() == size);
+    assert(m_nMeasurements.size() == size);
+    assert(m_nHoles.size() == size);
+
+    for (const auto& [key, col] : m_dynamic) {
+      (void)key;
+      assert(col->size() == size);
+    }
+  }
```

Comment:
is this temporary? I guess we can leave it as it should be optimized out in release build?

Reply:
So actually you're right that the `for` loop should probably go in case of a release build. Maybe I'll add function-level guards against this so it doesn't run at all.

Reply:
I guess the optimizer would remove it because otherwise the body is empty?

Reply:
Reworked this so the `assert` is (also) around the `checkConsistency` call, so suould be zero cost now.